### PR TITLE
Remove calls to empty RuntimeHelpers.PrepareConstrainedRegions()

### PR DESCRIPTION
**Pull request by Marusyk**
Remove calls to empty `RuntimeHelpers.PrepareConstrainedRegions()`

Issue #33179 

Skipped (please let me know if need some action on these):

- System.DirectoryServices.Protocols
- System.Diagnostics.EventLog
- System.Data.OleDb
- System.Data.Odbc
- [src/libraries/Common/src/System/Diagnostics/NetFrameworkUtils.cs](https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Diagnostics/NetFrameworkUtils.cs#L67)
- [src/installer/managed/Microsoft.NET.HostModel/AppHost/MachOUtils.cs](https://github.com/dotnet/runtime/blob/master/src/installer/managed/Microsoft.NET.HostModel/AppHost/MachOUtils.cs#L203)
- [src/coreclr/tests/src/GC/Performance/Tests/LowLatencyTest.cs](https://github.com/dotnet/runtime/blob/master/src/coreclr/tests/src/GC/Performance/Tests/LowLatencyTest.cs#L112)
- System.Reflection.Metadata - reverted

Please review,
Thank you in advance

### CODE REVIEW

File: src/libraries/System.Net.HttpListener/src/System/Net/Windows/WebSockets/WebSocketBase.cs [hunk 1]
```diff
@@ -1020,7 +1019,6 @@ private void ReleaseLocks(ref bool thisLockTaken, ref bool sessionHandleLockTake
 
             if (thisLockTaken || sessionHandleLockTaken)
             {
-                RuntimeHelpers.PrepareConstrainedRegions();
```
Comment (stephentoub) [inline src/libraries/System.Net.HttpListener/src/System/Net/Windows/WebSockets/WebSocketBase.cs]: You can remove the empty try block as well in such cases.

File: src/libraries/System.Private.CoreLib/src/System/IO/UnmanagedMemoryStream.cs [hunk 2]
```diff
@@ -128,7 +127,6 @@ protected void Initialize(SafeBuffer buffer, long offset, long length, FileAcces
             unsafe
             {
                 byte* pointer = null;
-                RuntimeHelpers.PrepareConstrainedRegions();
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/IO/UnmanagedMemoryStream.cs]: @akoeplinger, does mono still need these? If yes, anything under src/libraries/System.Private.Corelib should be reverted.

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/MemoryMappedFileBlock.cs [hunk 3]
```diff
@@ -18,10 +17,6 @@ private sealed class DisposableData : CriticalDisposableObject
 
             public DisposableData(IDisposable accessor, SafeBuffer safeBuffer, long offset)
             {
-                // Make sure the current thread isn't aborted in between acquiring the pointer and assigning the fields.
-#if !NETSTANDARD1_1
-                RuntimeHelpers.PrepareConstrainedRegions();
-#endif
```
Comment (stephentoub) [inline src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/MemoryMappedFileBlock.cs]: @tmat, I assume the netstandard2.0 build would still want these for .NET Framework? If yes, all of the changes under System.Reflection.Metadata should be reverted.

**Comment by stephentoub** [comment]
The ones under src/coreclr/src/System.Private.CoreLib/src should be fine to remove.

File: src/libraries/System.Private.CoreLib/src/System/IO/UnmanagedMemoryStream.cs (hunk 2 above)
Comment (akoeplinger) [reply src/libraries/System.Private.CoreLib/src/System/IO/UnmanagedMemoryStream.cs]: The implementation for `RuntimeHelpers.PrepareConstrainedRegions()` is empty so I don't think it's needed.

Comment (stephentoub) [reply src/libraries/System.Private.CoreLib/src/System/IO/UnmanagedMemoryStream.cs]: Ok, thanks, I wasn't sure if the VM recognized it specially as an intrinsic.

File: src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/MemoryMappedFileBlock.cs (hunk 3 above)
Comment (tmat) [reply src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/MemoryBlocks/MemoryMappedFileBlock.cs]: Yes, I would think so.

File: src/libraries/System.Net.HttpListener/src/System/Net/Windows/WebSockets/WebSocketBase.cs [hunk 4]
```diff
@@ -1020,23 +1019,16 @@ private void ReleaseLocks(ref bool thisLockTaken, ref bool sessionHandleLockTake
 
             if (thisLockTaken || sessionHandleLockTaken)
```
Comment (stephentoub) [inline src/libraries/System.Net.HttpListener/src/System/Net/Windows/WebSockets/WebSocketBase.cs]: This `if` is no longer necessary and can just be removed (the body should remain).

File: src/libraries/System.Net.HttpListener/src/System/Net/Windows/WebSockets/WebSocketBase.cs [hunk 5]
```diff
@@ -1104,7 +1096,6 @@ private static void ReleaseLock(object lockObject, ref bool lockTaken)
             Debug.Assert(lockObject != null, "'lockObject' MUST NOT be NULL.");
             if (lockTaken)
             {
-                RuntimeHelpers.PrepareConstrainedRegions();
```
Comment (stephentoub) [inline src/libraries/System.Net.HttpListener/src/System/Net/Windows/WebSockets/WebSocketBase.cs]: Another case where the empty try should be removed.

**Comment by Marusyk** [comment]
what is the reason of failed builds?

**Comment by stephentoub** [comment]
> what is the reason of failed builds?

A seg fault in a variety of coreclr tests on Linux.  I don't think it's related to your PR, but I'm running the tests again.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Comment by jkotas** [comment]
The Linux failures are known issue fixed by https://github.com/dotnet/runtime/pull/33580

**Pull request merged**